### PR TITLE
Add MaxMaxAgent

### PR DESCRIPTION
**Pull request by dlinman**

### CODE REVIEW

File: GameEngine/Agents/MaxMaxAgent.cs
Comment (rpherbig) [inline GameEngine/Agents/MaxMaxAgent.cs]: 💡 I'm not a fan of throwing a generic `Exception`. We've got a few custom ones but none seem to fit this case. What do you think about a new `NoMoveFoundException`?

**Review by rpherbig — APPROVED** [review]
💡 Idea that does not hold up merging this Pull Request.
❌ Issue that needs resolved before merging this Pull Request.

File: GameEngine/Agents/MaxMaxAgent.cs
Comment (DrSeabass) [reply GameEngine/Agents/MaxMaxAgent.cs]: Just for clarity's sake, this should never happen, right?

For there to be no legal moves, either the player has no cards or there are no owls not in nests.
If the player has no cards, all cards have been drawn.
If all cards have been drawn, all sun cards have been drawn.
If all sun cards have been drawn, the game is over as a loss.
On the other side, if all owls are in nests, the game is over as a win.

So, for the error to be raised, we should have to be trying to make a move after a win/loss condition in the game, which we shouldn't be doing.

Comment (dlinman) [reply GameEngine/Agents/MaxMaxAgent.cs]: Correct, this shouldn't be possible.

Comment (dlinman) [reply GameEngine/Agents/MaxMaxAgent.cs]: However, if someone messes with the configuration of the game, then it's possible that we could get into this state.  The null check is to make sure that if things change elsewhere in the codebase that this code continues to act in a well formed manner.  

If we end up returning a null then something someplace else will fail with a less than helpful null reference exception.  Failing here will make it a bit easier to figure out what's going wrong.

Comment (dlinman) [reply GameEngine/Agents/MaxMaxAgent.cs]: With respect to new exception type, sure 👍 .

**Pull request merged**
